pkg/chains/signing/wrap.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2021 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package signing
+
+import (
+	"context"
+	"crypto"
+	"encoding/json"
+
+	"github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/in-toto/in-toto-golang/pkg/ssl"
+)
+
+func Wrap(ctx context.Context, s Signer) (Signer, error) {
+	adapter := sslAdapter{
+		wrapped: s,
+	}
+
+	envelope, err := ssl.NewEnvelopeSigner(&adapter)
+	if err != nil {
+		return nil, err
+	}
+
+	pub, err := s.PublicKey(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &sslSigner{
+		wrapper: envelope,
+		typ:     s.Type(),
+		pub:     pub,
+	}, nil
+}
+
+// sslAdapter converts our signing objects into the type expected by the Envelope signer for wrapping.
+type sslAdapter struct {
+	wrapped Signer
+	KeyID   string
+}
+
+func (w *sslAdapter) Sign(data []byte) ([]byte, string, error) {
+	sig, _, err := w.wrapped.Sign(context.Background(), data)
+	return sig, w.KeyID, err
+}
+
+func (w *sslAdapter) Verify(keyID string, data, sig []byte) (bool, error) {
+	panic("unimplemented")
+}
+
+// sslSigner converts the EnvelopeSigners back into our types, after wrapping.
+type sslSigner struct {
+	wrapper *ssl.EnvelopeSigner
+	typ     string
```

Comment:
nit: should this be `type`?

Reply:
Hah, no this is intentional to annoy Jon and because type is a reserved word :)